src/wallet/wallet2.cpp

```diff
@@ -956,12 +956,17 @@ void wallet2::process_new_transaction(const crypto::hash &txid, const cryptonote
   }
 
   uint64_t tx_money_spent_in_ins = 0;
+  // deterministic txkey
+  std::vector<rct::key> deterministic_txkey_buf;
+  deterministic_txkey_buf.push_back(rct::sk2rct(m_account.get_keys().m_spend_secret_key));
   // check all outputs for spending (compare key images)
   for(auto& in: tx.vin)
   {
     if(in.type() != typeid(cryptonote::txin_to_key))
       continue;
-    auto it = m_key_images.find(boost::get<cryptonote::txin_to_key>(in).k_image);
+    crypto::key_image img = boost::get<cryptonote::txin_to_key>(in).k_image;
+    deterministic_txkey_buf.push_back(rct::ki2rct(img));
```

Comment:
Does this have any cost? Ideally we want to skip as many things as possible if we don't get a hit.

Reply:
It does have a cost of copying the key image to the buffer. I changed the code such that this additional processing is performed only when `use-deterministic-txkey` is enabled.